Report.java

```diff
@@ -34,31 +38,127 @@ public Report(Submission[] submissions)
      */
     public void makeReport()
     {
-        System.out.println("------------");
-        System.out.println("SCATT Report");
-        System.out.println("------------");
-        System.out.println();
-        for (int i = 0; i < submissions.length; i++)
+        File reportFile = new File("./Report.txt");
+        PrintWriter printW;
+        try
         {
-            if (submissions[i].isValid())
+            printW = new PrintWriter(reportFile);
+            printW.println("SCATT Report\n");
+            for (int i = 0; i < submissions.length; i++)
             {
-                System.out.println("File: " + submissions[i].getName());
-                System.out.println("---------------------------------");
-                System.out.println("Script count: " 
-                    + submissions[i].getScriptCount());
-                System.out.println("Sprite count: " 
-                    + submissions[i].getSpriteCount());
-                System.out.println();
-                String[] spriteNames = submissions[i].getSpriteNames();
-                for (int j = 0; j < spriteNames.length; j++)
+                if (submissions[i].isValid())
                 {
-                    System.out.println("Sprite: " + spriteNames[j]);
-                    System.out.println("Script count: " 
-                        + submissions[i].getScriptCountForSprite(
-                            spriteNames[j]));
-                    System.out.println();
+                    printTotalCounts(printW, i);
+
+                    printStageCounts(printW, i);
+
+                    printSpriteCounts(printW, i);
```

Comment:
Add a test - if the submission has no sprites, no need to call this (prints "Sprite Counts" with nothing beneath it and is a bit confusing).